### PR TITLE
PT-2323 Refactor REST to use the delete method from PatientRepository

### PR DESCRIPTION
**Pull request by AlexLapena**

### CODE REVIEW

File: components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java [hunk 1]
```diff
@@ -340,23 +338,19 @@ public void deletePatientCatchesXWikiException() throws XWikiException
...
-        XWiki wiki = mock(XWiki.class);
-        XWikiDocument patientXWikiDoc = mock(XWikiDocument.class);
-        doReturn(wiki).when(this.context).getWiki();
-        doReturn(patientXWikiDoc).when(this.patient).getXDocument();
         doReturn(true).when(this.access).hasAccess(Right.DELETE, this.userProfileDocument, this.patientDocument);
 
         Response response = this.patientResource.deletePatient(this.id);
 
-        verify(wiki).deleteDocument(patientXWikiDoc, this.context);
+        verify(this.logger).debug("Deleted patient record [{}]", this.id);
```
Comment (allasm) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java]: a minor point: I would rather check that no exceptions were thrown, or just removed this check and only left the check below (for the return type)

Comment (allasm) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java]: ...also not sure if we want to have this method be declared as `throws XWikiException`, maybe use `WebApplicationException` or catch all exceptions?

**Comment by AlexLapena** [comment]
Updated & rebased.

File: components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java [hunk 2]
```diff
@@ -340,23 +335,18 @@ public void deletePatientCatchesXWikiException() throws XWikiException
...
     {
-        XWiki wiki = mock(XWiki.class);
-        XWikiDocument patientXWikiDoc = mock(XWikiDocument.class);
-        doReturn(wiki).when(this.context).getWiki();
-        doReturn(patientXWikiDoc).when(this.patient).getXDocument();
         doReturn(true).when(this.access).hasAccess(Right.DELETE, this.userProfileDocument, this.patientDocument);
 
         Response response = this.patientResource.deletePatient(this.id);
 
-        verify(wiki).deleteDocument(patientXWikiDoc, this.context);
```
Comment (sdumitriu) [inline components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java]: You should replace this with a verification that the repository was invoked.

**Comment by sdumitriu** [comment]
You should do the same for the other resource, `DefaultPatientByExternalIdResourceImpl.java`.

**Comment by AlexLapena** [comment]
Added the change for the other resource.

**Pull request merged**